src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -1804,6 +1853,87 @@ def generate_points(self, rom_name: str, on_disk: bool = True, named_selection:
+        named_selection: str (optional)
+            Named selection on which the mesh projection has to be performed. The default is ``None``, in which case the
+            entire domain is considered.
+
+        Returns
+        -------
+        pyvista.DataSet
+            PyVista DataSet object of the targeted mesh with projected field ROM data.
+
+        Raises
+        ------
+        TwinModelError:
+            If ``TwinModel`` object has not been initialized.
+            If rom_name is not included in the Twin's list of TBROM
+            If TBROM hasn't its mode coefficients outputs connected to the twin's outputs
+            If mesh is not a valid grid dataset
+            If name_selection is not included in the TBROM's list of Named Selections
+            If interpolate is True and no points file is available with the TBROM
+
+        TwinModelWarning:
+            If interpolate is False and the targeted mesh has a number of cells and points different from TBROM point
+            cloud. In that case, interpolate is automatically switched to True.
+
+        Examples
+        --------
+        >>> from pytwin import TwinModel
+        >>> import pyvista as pv
+        >>> # Instantiate a twin model, initialize it, and evaluate it step by step until you want to save its state
+        >>> model1 = TwinModel('model.twin')
+        >>> model1.initialize_evaluation()
+        >>> romname = model1.tbrom_names[0]
+        >>> mesh = pv.read('mesh.vtk')
+        >>> fieldresults = model1.project_tbrom_on_mesh(romname, mesh)
+        """
+        self._log_key = "MeshProjection"
+
+        if not self.evaluation_is_initialized:
+            msg = self._error_msg_for_not_initialized()
+            self._raise_error(msg)
+
+        try:
+            if self._check_tbrom_mesh_projection_args(rom_name, mesh, named_selection):
+                if named_selection is None:
+                    nb_points = self._tbroms[rom_name].nb_points
+                else:
+                    nb_points = len(self._tbroms[rom_name].named_selection_indexes(named_selection))
+                if not interpolate and (mesh.n_cells != nb_points and mesh.n_points != nb_points):
+                    msg = (
+                        f"[MeshProjection]Switching interpolate flag from False to True. Number of TBROM points = "
+                        f"{nb_points}, number of mesh cells = {mesh.n_cells}, number of mesh points = {mesh.n_points}."
+                    )
+                    self._log_message(msg, PyTwinLogLevel.PYTWIN_LOG_WARNING)
+                    interpolate_flag = True
+                else:
+                    interpolate_flag = interpolate
+                if interpolate_flag:
+                    self._check_tbrom_points_file(rom_name)
+                self._tbroms[rom_name].project_on_mesh(mesh, interpolate_flag, named_selection)
+                self._update_tbrom_outmcs(self._tbroms[rom_name])
```

Comment:
Since we call the `._update_tbrom_outmcs` here, I would suggest to say in the docstring that the `project_on_mesh` method uses the last mode coordinates evaluated by the TwinModel.

Reply:
Also, I think we should explain the workflow in case user wants to perform many evaluations of the ROM, so it is clear that only one call to the `project_on_mesh` method must be done and that the field from the returned pyvista object is automatically updated each time we re-evaluate the TwinModel, which is the case right?

I'm talking about this workflow:
`twin = TwinModel("twin_with_tbrom")`
`twin.initialize_evaluation()`
`rom_result_on_mesh = twin.project_tbrom_on_mesh()`
Then performing new evaluation or initialization with the `twin` object will update output result from the `rom_result_on_mesh` object.

Reply:
- 1st comment - added (as well as in the get_tbrom_output_field method)
- 2nd additional information provided in the method description/docstrings as well as in the code snippet example (e.g. initialization is performed after the projection method used)